### PR TITLE
Implement support for Maya 2019

### PR DESCRIPTION
**Pull request by BigRoy**
This implements support for Maya 2019. This has been compiled on Windows 10 with Maya 2019.2 SDK and works as expected.

Please note however that the code has quite some refactored changes due to obsolete and deprecated functions in the Maya API in 2019. As such, I'm not sure whether this is still fully backwards compatible with Maya 2018 and below.

I'm opening up this PR as a basis for discussing the changes and also for others who might be interested in compiling for Maya 2019.

You should be able to build using Visual Studio 2017 with:

```
set MAYA_2019_SDK=C:/path/to/devkitBase
windows_build_plugin 2019
```

---

Note that somehow after compiling the files were *not* copied to my documents folder even though the successful compilation message does tell you it did it. This might be unrelated to this change or Maya 2019 as it's the first time I compiled the plug-in and have not done this for other Maya versions.

I took what was needed manually from the `build` folder, etc. and packaged it up in a Maya `.mod` file and folder so I could easily distribute it on a network.

### CODE REVIEW

**Comment by BigRoy** [comment]
I noticed some commits related to Maya 2019 having some case sensitivitiy issues which makes me guess that an implementation for Maya 2019 was already available. Anyway, if it is... sorry I missed it and feel free to close this PR. ;)

**Comment by Ziriax** [comment]
Thanks a lot for this PR! 

Yes a while ago I was working towards version 1.0 of this plugin in the `develop` branch, and was busy supporting Maya 2019.

I guess I missed the obsolete and deprecated and functions in the SDK, I need to investigate this.

We might have been targetting a different revision of Maya 2019 too.

I'm a bit busy this week, but will certainly look into this.

File: Maya2glTF.vcxproj [hunk 1]
```diff
@@ -40,6 +48,13 @@
     <CharacterSet>MultiByte</CharacterSet>
     <MayaVersion>2018</MayaVersion>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Maya2019-Debug|x64'" Label="Configuration">
+    <ConfigurationType>DynamicLibrary</ConfigurationType>
+    <UseDebugLibraries>true</UseDebugLibraries>
+    <PlatformToolset>v141</PlatformToolset>
+    <CharacterSet>MultiByte</CharacterSet>
+    <MayaVersion>2018</MayaVersion>
```
Comment (BigRoy) [inline Maya2glTF.vcxproj]: I just noticed this. I think these might still need to be refactored to 2019 even though compilation worked on my end. It's the same in the `Maya2019-Release|x64` property group.

**Comment by Ziriax** [comment]
I noticed indeed that when compiling my version of the 2019 plugin, I get a lot of warnings about deprecated functions. It is going to be a challenge to keep the plugin working for older Maya versions... 

Also, maybe it is time to stop supporting Maya 2016...

**Comment by stevetalkowski** [comment]
Hey Peter,

I’ve created a Dropbox folder for you here:

https://www.dropbox.com/sh/a75oj3uthtvty27/AACbOpbtTE8RL7WPPTdvYPEma?dl=0 <https://www.dropbox.com/sh/a75oj3uthtvty27/AACbOpbtTE8RL7WPPTdvYPEma?dl=0>

The folder "FBX2glTF_example" contains an FBX file that i used with the external Facebook FBX2glTF. When I drag the pilot_walkCycle_out folder onto <https://gltf-viewer.donmccurdy.com <https://gltf-viewer.donmccurdy.com/>> I can toggle and see the skeleton.

The folder “updated_test_2" contains the output from using the Maya2glTF plug in. I’m not seeing a skeleton output. Should the plugin have an option to include the skeleton, or does it detect one when the meshes are automatically selected?

Give those a try and let me know if you come across anything that I should try next.

Thanks!

-Steve

--

Steve Talkowski
Design. Create. Animate.
http://sketchbot.tv
@stevetalkowski

**Comment by Ziriax** [comment]
The `develop` branch now has support for Maya 2019, and will soon become the 1.0 master.

I looked at your changes and manually applied these on the `develop` branch, but I had to tweak it a bit so it worked with Maya 2017.

I'm going to drop support for Maya 2016, didn't try that one, it's too old.

Hope you don't mind I close this, thanks for your contribution!

**Pull request closed without merging**